src/Confix.Tool/src/Confix.Tool/Utilities/Dotnet/DotnetHelpers.cs

```diff
@@ -31,22 +34,60 @@ public static class DotnetHelpers
     public static FileInfo? GetAssemblyFileFromCsproj(FileInfo projectFile)
     {
         // Load the .csproj file as an XDocument
-        var csprojDoc = XDocument.Load(projectFile.FullName);
-
-        XNamespace msbuild = "http://schemas.microsoft.com/developer/msbuild/2003";
+        var csprojDoc = XDocument.Load(projectFile.FullName, LoadOptions.PreserveWhitespace);
 
         // Extract the assembly name and target framework from the .csproj file
         var propertyGroup =
-            csprojDoc.Element(msbuild + "Project")
-                ?.Element(msbuild + "PropertyGroup")
-                ?.Element(msbuild + "AssemblyName")
+            csprojDoc.Element(Xml.Project)
+                ?.Element(Xml.PropertyGroup)
+                ?.Element(Xml.AssemblyName)
                 ?.Value ??
             Path.GetFileNameWithoutExtension(projectFile.FullName);
 
         // Construct the path to where the assembly should be built
         return GetAssemblyInPathByName(projectFile.Directory!, propertyGroup);
     }
 
+    public static string EnsureUserSecretsId(FileInfo csprojFile)
+    {
+        App.Log.EnsuringUserSecretsIdInTheCsprojFile(csprojFile.FullName);
+
+        var csprojDoc = XDocument.Load(csprojFile.FullName, LoadOptions.PreserveWhitespace);
+
+        var project = csprojDoc.Element(Xml.Project);
+        if (project == null)
+        {
+            App.Log.CsprojFileDoesNotContainAProjectElement(csprojFile.FullName);
+            project = new XElement(Xml.Project);
+            csprojDoc.Add(project);
+        }
+
+        var propertyGroup = project.Elements(Xml.PropertyGroup).FirstOrDefault();
```

Comment:
In theory you can have multiple Property Groups. in this case the implementation would fore it in the first one. Is that desired?